lib/kms_client.rb

```diff
@@ -16,7 +16,16 @@ def decrypt(cipher)
   end
 
   def self.aws_kms_client
-    @@kms = Aws::KMS::Client.new(region: 'us-east-1', stub_responses: ENV['APP_ENV'] == 'test') if @@kms.nil?
+    if ENV['LOCAL']
+      @@kms = Aws::KMS::Client.new(
+        region: 'us-east-1',
+        stub_responses: ENV['APP_ENV'] == 'test',
+        access_key_id:  ENV['AWS_ACCESS_KEY_ID'],
+        secret_access_key: ENV['AWS_SECRET_ACCESS_KEY']
```

Comment:
This will fail when deployed because these env vars aren't set in deployed code.

Reply:
Unless I am misunderstanding, this will not be called in deployed code since `ENV['LOCAL']` is not set

Reply:
Ah - I didn't see the `ENV['LOCAL']` check. I think we should try to keep differences between local and remote invocations to a minimum. The `region` and `stub_responses` params may be identical in either case, so that leaves the creds. I suspect the auth issue you were running into with `sam` and `--profile` may be resolved [like this](https://github.com/NYPL/recap-hold-request-consumer/blob/2a4d4b8188c8a0d4295f5171d62d7ad640040eab/models/kms.rb#L9C22-L10). Essentially, there's an old bug in the Docker/AWS interaction that creates a nonsensical `AWS_SESSION_TOKEN` of "", which confuses auth. Since that value should never be blank legitimately, it's safe to delete it when blank. Having added that, you should find you can run this with `sam` and `--profile` locally using only `@@kms = Aws::KMS::Client.new(region: 'us-east-1', stub_responses: ENV['APP_ENV'] == 'test') if @@kms.nil?`

Reply:
Interesting. Looks like that does work. I've updated this PR, will also post a PR to change this in the `sync-item-metadata-to-scsb-service`